src/poprox_recommender/default.py

```diff
@@ -82,18 +99,26 @@ def personalized_pipeline(num_slots: int, algo_params: dict[str, Any] | None = N
         logger.info("Recommendations will be ranked with plain top-k.")
         ranker = topk_ranker
 
-    pipeline = RecommendationPipeline(name=diversify)
-    pipeline.add(article_embedder, inputs=["candidate"], output="candidate")
-    pipeline.add(article_embedder, inputs=["clicked"], output="clicked")
-    pipeline.add(user_embedder, inputs=["clicked", "profile"], output="profile")
-    pipeline.add(article_scorer, inputs=["candidate", "profile"], output="candidate")
-    pipeline.add(topk_ranker, inputs=["candidate", "profile"], output="ranked")
-    pipeline.add(ranker, inputs=["candidate", "profile"], output="reranked")
+    pipeline = Pipeline()
+    candidates = pipeline.create_input("candidate", ArticleSet)
+    clicked = pipeline.create_input("clicked", ArticleSet)
+    profile = pipeline.create_input("profile", InterestProfile)
+    e_cand = pipeline.add_component("embed-candidates", article_embedder, article_set=candidates)
```

Comment:
The verb style naming here feels a little weird to me, because a component is conceptually a thing rather than an action. If you really want the verb style, I don't think `add_component` is the right method name to go with that (but I do like `add_component` because I think it aligns more naturally with the mental model that e.g. a diversifier is a component of a larger pipeline/system.)

Reply:
Fair! I was trying to do something consistent (and make a consistent recommendation in the docs), and wasn't 100% on this but favored “pick one, be consistent, and document the rec”.

We can use nouns / noun phrases, and it would be slightly longer in some cases but I think would read reasonably well:

```
e_cand = pipeline.add_component("candidate-embedder", ...)
```

Or if we want to name it after the thing returned:

```
e_cand = pipeline.add_component("candidate-embeddings", ...)
```

Working through that example, we have 3 possible naming conventions:

1. The name of the component itself (embedder).
2. The name of the component's return data (embeddings).
3. The action performed by the component (embed).

Based on your rationale, it sounds like you would lean towards (1).

Reply:
I've pushed an update with the component name naming.

Reply:
Would it make more sense to have something like `add_process` or `add_stage` and use the verb-based naming? Not sure it would, just pondering while I update the docs.

Reply:
I like the noun-style naming better because I find it easier to think about composing objects than actions, but if we went the verb-style route I was considering `add_step`